src/main/java/org/apache/sling/event/impl/jobs/scheduling/JobSchedulerImpl.java

```diff
@@ -455,6 +455,14 @@ private boolean match(final ScheduledJobInfoImpl job, final Map<String, Object>
         }
         return jobs;
     }
+    
+    /**
+     * Provide the total number of jobs registered in the system, irrespective of topics
+     * @return the total number of scheduled jobs
+     */
+    public int getTotalNumberOfScheduledJobs() {
+       return this.scheduledJobs.size(); 
```

Comment:
sure, to deliver precise results it would need synchronization. But this is "just" metrics, so not always 100% accurate would be totally ok.
On the other hand I the overhead is probably very minor, because it is expected that this map does not change a lot.

I will add this synchronization.

Reply:
good point, it's a bit of overly cautious to make it synchronous indeed .. oh well, iiuc this is mostly not synchronized unless a new schedule is added or removed. so it should indeed not impact noticeably